train.py

```diff
@@ -50,47 +55,118 @@ def write_info_file():
         np.savetxt(args.outf+'/samples/output' + str(image) + '.txt', output)
-        truth = data.numpy()
-        noisy = noisy.numpy()
-        diff = output-truth
-        noisy_diff = noisy-truth
+        sharp = dataSharp.numpy()
+        fuzzy = dataFuzz.numpy()
+        diff = output-sharp
+        fuzzy_diff = fuzzy-sharp
         np.savetxt(args.outf+'/samples/diff' + str(image) + '.txt', diff)
-        del data
-        del noisy
+        del sharp
+        del fuzzy
         del output
         del diff
     model.to('cuda')
 
+
+#gets data needed to make histograms of sample data
+#assumes that the xmin, xmax, ymin, and ymax values are the same 
+#for all four high and low quality training and validation files
+file = up.open(args.valfileSharp)
+tree = file["g4SimHits/tree"]
+x_min = tree["xmin"].array().to_numpy()[0]
+x_max = tree["xmax"].array().to_numpy()[0]
+y_min = tree["ymin"].array().to_numpy()[0]
+y_max = tree["ymax"].array().to_numpy()[0]
+x_bins = tree["xbins"].array().to_numpy()[0]
+y_bins = tree["ybins"].array().to_numpy()[0]
+
+#makes histograms given bin weights listed in .txt file    
+def make_plots(fin, data):
+    binweights = np.loadtxt(fin)
+    binarray = []
+    for i, elem in enumerate(binweights):
+        for j, elem in enumerate(binweights[i]):
+            binarray.append(binweights[i][j])
+            
+    if 'output' in fin:
+        event_number = fin.replace(args.outf+'/samples/'+'output','')
+        event_number = event_number.replace('.txt','')
+        event_number = int(event_number)
+        means = np.mean(data[0][0].numpy())
+        stdevs = np.std(data[0][0].numpy())
+        
+        for i, elem in enumerate(binarray):
+            binarray[i] *= stdevs
+            binarray[i] += means
+        
+    else:
+        print('not output') 
+        
+    #builds axes for histogram given min/max and bin number
+    x_axis = []
+    count = 0
+    x_start = x_min
+    while count < 100:
+        for i in range(100):
+            x = x_start + ((x_max-x_min)/x_bins)*float(i)
+            x_axis.append(x)
+        count = count + 1
+
+    y_axis = []
+    y_start = y_min
+    for i in range(100):
+        y = y_start + ((y_max-y_min)/y_bins)*float(i)
+        count = 0
+        while count < 100:
+            y_axis.append(y)
+            count = count + 1
+            
+    #makes histogram
+    fig = plt.subplots(figsize =(10, 7))
+    plt.hist2d(x_axis, y_axis, bins=[x_bins,y_bins], weights = binarray)
+    plt.title("Energy Deposits Projected on z plane")
+    plt.xlabel("x (cm)")
+    plt.ylabel("y (cm)")
+    plt.colorbar(label = "Energy (MeV)")
+    fin = str(fin).replace(args.outf+'/samples/','')
+    plt.savefig(args.outf+'/plots/' + str(fin).replace( '.txt', '.png'))
+
 def init_weights(m):
     if type(m) == nn.Linear:
         torch.nn.init.xavier_uniform(m.weight)
         m.bias.data.fill_(0.01)
 
 def main():
+    random.seed(args.randomseed)
```

Comment:
you should add `torch.manual_seed(args.randomseed)` to make the training reproducible

Reply:
```suggestion
    random.seed(args.randomseed)
    torch.manual_seed(args.randomseed)
```
I'm not entirely sure what randomness we are seeding with this (initial values for output data before going through CNN)?

Reply:
the torch random seed controls the initial values of the weights in the CNN before training (among other things)